### PR TITLE
post links

### PR DESCRIPTION
**Pull request by cindywu**
The post links functionality listed below is in this PR and not breaking. The feature needs polish.

https://www.loom.com/share/2cbe717d801d44608c297103cc0cc69c

- [x] toolbar icon → forminput form → all posts
- [x] input form →function ()→links or generates posts
- [x] backend: links posts /link
- [x] backend: creating posts /generate
- [x] ranges can always add new links
- [x] hover portal: delete link
- [x] hover portal: title
- [x] showing list of post links, making it scrollable

Polish:

- [x] add user_id attached to new post from post link plugin
- [x] fix the generated post editor breaking
- [x] dedupe when same post is added multiple times
- [x] add a new postlink icon
- [x] fix new post editor is blank
- [x] styles for post link ui
- [x] sanitize title for preview in post seralizer
- [x] if no title for post, show indicator that there is no title
- [x] fix title url in renderPostLink 
- [ ] do not scroll user to top when showing post link search bar
- [ ] truncate title for search results
- [ ] fix having to refresh show title of post link
- [ ] refactor `PostLinkSearch.js` to a function component

Research:

- for search results, do we prioritize number of results or more infomormation about the result? answer is influenced by the accuracy and speed of our search algorithm. if very accurate, then prioritize search result data. if not accurate, prioritize showing more results. 
- when in post link form determine how to make the form reactive to whether or not user is generating a post or linking to existing post. option 1: show label that is linking or generating option 2: dynamic button text option 3: question mark button to guide
- adding a url as a post link/reference

### CODE REVIEW

**Comment by dml-1** [comment]
can you remove the 3 .ds_store files that were checked in and then push again?

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -39,7 +39,7 @@ def create
     respond_to do |format|
       if @post.save
         format.html { redirect_to @post, notice: 'Post was successfully created.' }
-        format.json { render json: { redirect_to: post_url(@post) } }
+        format.json { render json: { redirect_to: post_url(@post), post: @post.as_json } }
```
Comment (dml-1) [inline app/controllers/posts_controller.rb:42]: this should return a new `@post.seralize` instead.

File: config/routes.rb [hunk 2]
```diff
@@ -55,6 +55,10 @@
   post '/add_comment', to: 'comments#create', as: :add_comment
   post '/remove_comment', to: 'comments#delete', as: :remove_comment
 
+  resources :citations
+  post '/add_citation', to: 'citations#add_citation', as: :add_citation
+  post '/remove_citation', to: 'citations#delete', as: :remove_citation
```
Comment (dml-1) [inline config/routes.rb:60]: these should be named consistently.

Comment (cindywu) [reply config/routes.rb:60]: i don't think i understand the feedback here. are you trying to say that throughout the app we use citation, post link, and reference interchangeably and we should stick to one of the three?

Comment (dml-1) [reply config/routes.rb:60]: the action name should be #remove_citation

**Pull request merged**